mongo/integration/server_selection_prose_test.go

```diff
@@ -79,34 +172,12 @@ func TestServerSelectionProse(t *testing.T) {
 				break
 			}
 		}
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
 
-		// Start 10 goroutines that each run 10 findOne operations.
-		var wg sync.WaitGroup
-		for i := 0; i < 10; i++ {
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
-
-				for i := 0; i < 10; i++ {
-					res := mt.Coll.FindOne(context.Background(), bson.D{})
-					assert.NoError(t, res.Err(), "FindOne() error")
-				}
-			}()
-		}
-		wg.Wait()
-
-		// Get all checkOut events and calculate the number of times each server was selected. The
-		// prose test spec says to use command monitoring events, but those don't include the server
-		// address, so use checkOut events instead.
-		checkOutEvents := tpm.Events(func(evt *event.PoolEvent) bool {
-			return evt.Type == event.GetStarted
-		})
-		counts := make(map[string]int)
-		for _, evt := range checkOutEvents {
-			counts[evt.Address]++
-		}
-		assert.Equal(mt, 2, len(counts), "expected exactly 2 server addresses")
+		awaitSaturation(ctx, mt, tpm, maxPoolSize)
```

Comment:
```suggestion
		awaitSaturation(ctx, mt, tpm, maxPoolSize * 2)
```

There will be maxPoolSize connections created per host.

Reply:
@kevinAlbs This should be taken care of by this line: https://github.com/mongodb/mongo-go-driver/pull/943/files/ac445db64c270ced70c858130e31862a75dce263#diff-ae8814df53c89fb89e6578e72eeeb1ef089f9307404f73f953b1f4e6ff7bc97aR48

Reply:
I suggest removing the multiplier from `isUnsaturated`. Or pass in a separate `numHosts` argument.

`len(options.Client().ApplyURI(mtest.ClusterURI()).Hosts)` may be more than 2. These tests override the number of hosts to exactly 2.

Reply:
I've implemented a solution that will see if a tolerable number of connections have been made for each host on the cluster URI.  In pseudo-code this looks like:

```
FOR host IN hosts
    // make sure no hosts below the tolerance
    IF host.connections < tolerance
        RETURN false
RETURN true
```

I think this may help add guard rails to `awaitSaturation`, where the count input is potentially ambiguous.

Reply:
New proposal LGTM.